### PR TITLE
Add openshift-metal3/terraform-provider-ironic repo prow integration

### PR DESCRIPTION
**Pull request by ardaguclu**
This PR adds required configuration to run prow configurations
in openshift-metal3/terraform-provider-ironic repository.

### CODE REVIEW

File: ci-operator/config/openshift-metal3/terraform-provider-ironic/openshift-metal3-terraform-provider-ironic-master.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+base_images:
+  base:
+    name: "4.10"
+    namespace: ocp
+    tag: base
+  dev-scripts:
+    name: test
+    namespace: ocp-kni
+    tag: dev-scripts
+  ocp_builder_rhel-8-golang-1.16-openshift-4.10:
```
Comment (elfosardo) [inline ci-operator/config/openshift-metal3/terraform-provider-ironic/openshift-metal3-terraform-provider-ironic-master.yaml:10]: let's use golang 1.17 here and below

**Review by elfosardo — CHANGES_REQUESTED** [review]

File: ci-operator/config/openshift-metal3/terraform-provider-ironic/openshift-metal3-terraform-provider-ironic-master.yaml (hunk 1 above)
Comment (ardaguclu) [reply ci-operator/config/openshift-metal3/terraform-provider-ironic/openshift-metal3-terraform-provider-ironic-master.yaml:10]: Good point, thanks @elfosardo

File: ci-operator/config/openshift-metal3/terraform-provider-ironic/openshift-metal3-terraform-provider-ironic-master.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    namespace: ocp
+    tag: rhel-8-golang-1.17-openshift-4.10
+binary_build_commands: make install
+build_root:
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: golang-1.17
+promotion:
+  name: test
```
Comment (andfasano) [inline ci-operator/config/openshift-metal3/terraform-provider-ironic/openshift-metal3-terraform-provider-ironic-master.yaml:21]: What's the reason for having a promote section here?

Comment (ardaguclu) [reply ci-operator/config/openshift-metal3/terraform-provider-ironic/openshift-metal3-terraform-provider-ironic-master.yaml:21]: No specific reason, probably it came with while I copied from some other repo :) and after you asking this maybe it is redundant?

Comment (andfasano) [reply ci-operator/config/openshift-metal3/terraform-provider-ironic/openshift-metal3-terraform-provider-ironic-master.yaml:21]: Yes, the `ocp-kni/test` it's currently just used to store the `dev-scripts` image (as it is not part of the ocp payload, but still required by some jobs), so better to remote this section

**Comment by ardaguclu** [comment]
@andfasano @elfosardo could you PTAL one more time to this PR?, thanks.

File: ci-operator/config/openshift-metal3/terraform-provider-ironic/openshift-metal3-terraform-provider-ironic-master.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+base_images:
+  base:
+    name: "4.10"
```
Comment (andfasano) [inline ci-operator/config/openshift-metal3/terraform-provider-ironic/openshift-metal3-terraform-provider-ironic-master.yaml:3]: Probably now that we are the past the code freeze, all the 4.10 references could be directly updated to 4.11?

**Pull request merged**